src/Turtle/Prelude.hs

```diff
@@ -958,6 +958,7 @@ grep pattern s = do
 -}
 sed :: Pattern Text -> Shell Text -> Shell Text
 sed pattern s = do
+    when (matchesEmpty pattern) (die "sed was passed a pattern which matches the empty string")
```

Comment:
Also, maybe change this error message to `"sed: the given pattern matches the empty string"`